pkg/network/tracer/tracer_test.go

```diff
@@ -976,21 +976,23 @@ func (s *UDPServer) Run(payloadSize int) error {
 		networkType = s.network
 	}
 	var err error
+	var ln net.PacketConn
 	if s.lc != nil {
-		s.ln, err = s.lc.ListenPacket(context.Background(), networkType, s.address)
+		ln, err = s.lc.ListenPacket(context.Background(), networkType, s.address)
 	} else {
-		s.ln, err = net.ListenPacket(networkType, s.address)
+		ln, err = net.ListenPacket(networkType, s.address)
 	}
 	if err != nil {
 		return err
 	}
 
+	s.ln = ln
 	s.address = s.ln.LocalAddr().String()
 
 	go func() {
 		buf := make([]byte, payloadSize)
 		for {
-			n, addr, err := s.ln.ReadFrom(buf)
+			n, addr, err := ln.ReadFrom(buf)
```

Comment:
Just curious what motivated the changes in this file? They seem to be functionally identical.

Reply:
`s.ln` is set to `nil` in `UDPServer.Shutdown`; this was causing a crash here.